addons/binding/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/handler/HDPowerViewHubHandler.java

```diff
@@ -0,0 +1,189 @@
+        webTargets = new HDPowerViewWebTargets(client, config.ipAddress);
+        refreshInterval = config.refresh;
+
+        updateStatus(ThingStatus.OFFLINE);
+
+        schedulePoll();
+
+        // Note: When initialization can NOT be done set the status with more details for further
+        // analysis. See also class ThingStatusDetail for all available status details.
+        // Add a description to give user information to understand why thing does not work
+        // as expected. E.g.
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+        // "Can not access device as username and/or password are invalid");
+    }
+
+    public HDPowerViewWebTargets getWebTargets() {
+        return webTargets;
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    void pollNow() {
+        schedulePoll();
+    }
+
+    private void schedulePoll() {
+        if (pollFuture != null) {
+            pollFuture.cancel(false);
+        }
+        pollFuture = scheduler.scheduleAtFixedRate(pollingRunnable, 500, refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private synchronized void poll() {
+        try {
+            Shades shades = webTargets.getShades();
+            if (shades != null) {
+                Map<Integer, Thing> things = getThingsByShadeId();
+                for (Shade shade : shades.shadeData) {
+                    Thing thing = things.get(shade.id);
+                    if (thing != null) {
+                        HDPowerViewShadeHandler handler = ((HDPowerViewShadeHandler) thing.getHandler());
+                        if (handler != null) {
+                            handler.onReceiveUpdate(shade);
+                        }
+                    }
+                }
+                updateStatus(ThingStatus.ONLINE);
+            }
+            Scenes scenes = webTargets.getScenes();
+            if (scenes != null) {
+                Map<Integer, Thing> things = getThingsBySceneId();
+                for (Scene scene : scenes.sceneData) {
+                    Thing thing = things.get(scene.id);
+                    if (thing != null) {
+                        HDPowerViewSceneHandler handler = ((HDPowerViewSceneHandler) thing.getHandler());
+                        if (handler != null) {
+                            handler.setOnline();
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.warn("Could not connect to bridge", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, e.getMessage());
```

Comment:
Are you sure that the exception is because the bridge is offline? If there can be different reasons, better catch different exceptions or remove the status detail and only keep the message.